Add a pydax mvp notebook as an example

edwardleardi opened the pull request:
The notebook pins the pydax install version so we don't have to worry
about too much breaking

We can delete this before the first release when we add more user guides

---

## Checklist:

<details open>

<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [ ] No
    - [ ] The change in this pull request is minor.
  - [x] Yes
    - Close https://github.ibm.com/CODAIT/CODAIT-Machine-Learning-Team/issues/717

For the following questions, only check the boxes that are applicable.

- [x] Ensure that the pull request text above the horizontal line is descriptive.
- [x] Add/update relevant tests.
- [x] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [ ] No, this pull request also acts as an issue by itself.
        - [ ] Assign yourself.
        - [ ] Label this pull request.
        - [ ] Put a milestone.
        - [ ] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [ ] Put this pull request under the appropriate epic on ZenHub.
        - [ ] Put an estimate on ZenHub.
        - [ ] Add dependency relationship with other issues/pull requests on ZenHub.
      - [x] Yes, this pull request addresses an issue and does not act as an issue.
        - [ ] Connect this pull request to the underlying issue on ZenHub.
        - [x] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [ ] When the pull request is ready, request review.
</details>

xuhdev reviewed (commented):
How about putting it in `docs/notebooks`? Examples will be throughout the document.

Just to keep the git history from being clustered, can we let git treat this file as binary? (Same for the key and pem files in `tests/tls/`, actually.) Technically we only deal with these files with wholesale changes (detailed line changes aren't interesting for these files), and any two non-trivial PRs make changes to these files should be treated as conflicting, no matter what lines they are touching (because the whole files are integrated, e.g., a single cell change would also update a lot of metadata). You can do this by following: https://stackoverflow.com/a/11162362/1150462

xuhdev commented on examples/pydax-mvp-demo.ipynb:
```diff
@@ -0,0 +1,1032 @@
+{
```
<p>I would insert two cells above this cell that runs:</p><p> </p><pre class="ql-syntax" spellcheck="false">Load a dataset is simple: One single line you'll have a remote dataset readily loaded, no hassle!
</pre><p> </p><pre class="ql-syntax" spellcheck="false">dataset = pydax.load_dataset('wikitext103')  # Load a dataset
dataset['valid'][:730]  # preview my dataset
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/pydax/pull/106/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

xuhdev replied:
<p>Actually if possible this can be put at the top. The most important thing is to get people to be interested first, then we let them learn a bit expanded basic usage/highlighted features to lock in their interest. After that, they should be able to read the document and figure out advanced features. But this might be better done in the landing page of our document later.</p>

edwardleardi committed:
Add a pydax mvp notebook as an example

The notebook pins the pydax install version so we don't have to worry
about too much breaking

We can delete this before the first release when we add more user guides

edwardleardi commented on docs/notebooks/pydax-mvp-demo.ipynb:
```diff
@@ -0,0 +1,1039 @@
+{
```
<p>I expanded the example in the first markdown cell. There's no clean way to run it as a code cell since we need to install pydax before running it so I just embedded the output along with the code as markdown. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/pydax/pull/106/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

edwardleardi committed:
Merge branch 'master' into mvp-notebook

xuhdev reviewed (approved):
Looks good!
